share/cmake/modules/FindExpat.cmake

```diff
@@ -59,12 +59,20 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
     # Attempt to find static library first if this is set
     if(EXPAT_STATIC_LIBRARY)
         set(_EXPAT_STATIC "${CMAKE_STATIC_LIBRARY_PREFIX}expat${CMAKE_STATIC_LIBRARY_SUFFIX}")
```

Comment:
I restructured the lib name list construction. Hopefully it's more clear now. Since this part of the code is creating a prioritised list of lib names to search for that *might* already be installed on the system, it needs to handle more than one case. i.e. not failing if the user has a Release version of a dependency, even though they are building OCIO in Debug. The code prioritises Debug libs in this case, but it *can* fallback to other variations.